src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,155 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> header;
+    private Node<T> tail;
+    private int size;
+
+    private class Node<T> {
+        private T value;
+        private Node<T> next;
+        private Node<T> prev;
+
+        public Node(Node<T> prev, T value, Node<T> next) {
+            this.prev = prev;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
     @Override
     public boolean add(T value) {
-        return false;
+        Node<T> newNode = new Node<>(tail, value, null);
+        if (size == 0) {
+            header = newNode;
+        } else {
+            tail.next = newNode;
+        }
+        tail = newNode;
+        size++;
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Index is out of bounds");
+        }
```

Comment:
this block can be like checkIndex method, but you should use it after next block (40-42), when you have already checked the condition if (index == size), you use checkIndex(index)

Reply:
done 
now I understood why 2 test-cases were crushed))